File: merxly_backend/merxly.Domain/Entities/Product.cs
using merxly.Domain.Interfaces;

namespace merxly.Domain.Entities
{
    public class Product : ICreatedDate, IModifiedDate
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public bool IsStoreFeatured { get; set; } // A flag to mark products as Feature Product
        public bool IsPlatformFeatured { get; set; }
        public bool IsActive { get; set; }
        public decimal? MinPrice { get; set; }
        public decimal? MaxPrice { get; set; }
        public int TotalStock { get; set; }
        public string? MainMediaPublicId { get; set; }
        public double AverageRating { get; set; }
        public int ReviewCount { get; set; }
        public int TotalSold { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        // Foreign Keys
        public Guid CategoryId { get; set; }
        public Guid StoreId { get; set; }

        // Navigation properties
        public Category Category { get; set; }
        public Store Store { get; set; }
        public ICollection<ProductAttribute> ProductAttributes { get; set; } = new List<ProductAttribute>();
        public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
    }
}

File: merxly_frontend/src/types/models/product.ts
import type { StoreProductSortBy, StoreProductSortOrder } from '../enums';
import type {
  CreateProductAttributeDto,
  ProductAttributeDto,
} from './productAttribute';
import type {
  CreateProductVariantDto,
  ProductVariantDto,
} from './productVariant';

export interface ProductDto {
  id: string;
  name: string;
  minPrice: number | null;
  maxPrice: number | null;
  totalStock: number;
  mainMediaPublicId: string | null;
  averageRating: number;
  reviewCount: number;
  isStoreFeatured: boolean;
  isPlatformFeatured: boolean;
  totalSold: number;
  categoryName: string;
  categoryId: string;
  createdAt: string;
}

export interface ProductForStore {
  id: string;
  name: string;
  isStoreFeatured: boolean;
  isActive: boolean;
  totalStock: number;
  totalVariants: number;
  mainMediaUrl: string | null;
  createdAt: string;
  updatedAt: string | null;
  categoryId: string;
  categoryName: string;
}

export interface ProductQueryParameters {
  pageNumber?: number;
  pageSize?: number;
  isStoreFeatured?: boolean;
  isActive?: boolean;
  categoryId?: string;
  searchTerm?: string;
  sortBy?: StoreProductSortBy;
  sortOrder?: StoreProductSortOrder;
}

export interface CreateProductDto {
  name: string;
  description: string | null;
  isStoreFeatured: boolean;
  isActive: boolean;
  categoryId: string;
  productAttributes: CreateProductAttributeDto[];
  variants: CreateProductVariantDto[];
}

export interface StoreDetailProductDto {
  id: string;
  name: string;
  description: string | null;
  isStoreFeatured: boolean;
  isActive: boolean;
  averageRating: number;
  reviewCount: number;
  totalSold: number;
  categoryId: string;
  categoryName: string;
  productAttributes: ProductAttributeDto[];
  variants: ProductVariantDto[];
}

export interface UpdateProductDto {
  name?: string;
  description?: string | null;
  isStoreFeatured?: boolean;
  isActive?: boolean;
  categoryId?: string;
}

export interface ResponseUpdateProductDto {
  id: string;
  name: string;
  description: string | null;
  isStoreFeatured: boolean;
  isActive: boolean;
  categoryId: string;
}
